filesystems/nfs/connectathon/runtest.sh

```diff
@@ -0,0 +1,561 @@
+. /mnt/tests/kernel/include/runtest.sh
+. kvercmp.sh
+
+# Check the test is run via CI. If it is, we should make sure task param
+# which is needed for kernel-ci is added, e.g.
+#     <params>
+#         <param name="CI" value="yes"/>
+#     </params>
+function is_run_byci ()
+{
+    [[ $CI == "yes" ]] && return 0 || return 1
+}
+
+
+# Check command /usr/bin/rup does exist because package 'rusers'
+# is no longer available on RHEL 8
+function check_cmd_rup ()
+{
+    [[ -x /usr/bin/rup ]] && return 0
+    echo "Oops, /usr/bin/rup not found" >&2
+    echo "$(uname -a)" >&2
+    return 1
+}
+
+function getServerList ()
+{
+    # Determine NFS server list
+    # NOTE: Beaker sets the NFSSERVERS environment variable in the file
+    # /etc/profile.d/rh-env.sh which is generated by a %post script in
+    # the kickstart file.
+    echo " ========== Determine NFS server list =========="
+
+    if [[ -z $NFSSERVERS ]]; then
+        echo " ========== Use static NFS server list =========="
+        servers="rhel3-nfs:/export/home rhel4-nfs:/export/home rhel5-nfs:/export/home rhel6-nfs:/export/home rhel7-nfs:/export/home netapp-v41:/export/kernel-test netapp-nfs:/vol/vol0/home/cthon sol10-nfs:/export/home"
+    else
+        echo " ========== Use dynamic NFS server list =========="
+        servers=$NFSSERVERS
+    fi
+}
+
+function checkServers ()
+{
+    local foundserver=0
+    local offlineserver=0
+    local thisdom=`dnsdomainname`
+
+    echo "" | tee -a $OUTPUTFILE
+    echo " ========== Confirming NFS servers in domain are online ==========" | tee -a $OUTPUTFILE
+    echo " ***** INFO: Current domain is \"$thisdom\" *****" | tee -a $OUTPUTFILE
+
+    for S in $servers; do
+        local server=$(echo $S| cut -f1 -d:)
+        ping -c 3 $server >> $OUTPUTFILE 2>&1
+        if [ $? != 0 ]; then
+            echo " ***** Warn: $server not online or unavailable in this domain *****" | tee -a $OUTPUTFILE
+            offlineserver=$(expr "$offlineserver" + 1)
+        else
+            echo " ***** Confirmed: $server is online *****" | tee -a $OUTPUTFILE
+            foundserver=$(expr "$foundserver" + 1)
+        fi
+    done
+
+    if [[ $foundserver -gt 0 && $offlineserver -eq 0 ]]; then
+        echo "" | tee -a $OUTPUTFILE
+        echo " ***** INFO: All servers in domain are online *****" | tee -a $OUTPUTFILE
+        echo " ========== Continue testing... ==========" | tee -a $OUTPUTFILE
+        echo "" | tee -a $OUTPUTFILE
+
+    elif [[ $foundserver -gt 0 && $offlineserver -gt 0 ]]; then
+        echo "" | tee -a $OUTPUTFILE
+        echo " ***** Warn: Some servers are offline or unavailable in this domain *****" | tee -a $OUTPUTFILE
+        echo " ========== Continue testing with online servers only... ==========" | tee -a $OUTPUTFILE
+        echo "" | tee -a $OUTPUTFILE
+        # Lets actually warn the test here and investigate, as we will have incomplete testing results
+        report_result $TEST WARN/ABORTED
```

Comment:
I'm confused here. Can you actually merge the result and status to be reported this way? ccing @rasibley who may have more insight

Reply:
@veruu @idorax I didn't think so, but not 100% valid beaker results are PASS, WARN, FAIL, or SKIP. Then use rhts-abort -t recipe to abort the test, https://beaker-project.org/docs/user-guide/task-environment.html 

It might be nice to fake out the test to actually warn/abort on one of these scenarios to see that it's represented correctly in beaker :-)

Reply:
@idorax can you take a look?

Reply:
@veruu @rasibley I checked the whole runtest.sh, it checked all servers are online one by one via function checkServers() before invoking cthon_main(), and cthon_main() check the server for testing is oneline again. To make the logic simple, I'd like to trust the results checked by checkServers(), and don't check the server (IPv4) is pingable again in cthon_main(), but keep the logic to check IPv6. Does this make sense?

Reply:
@veruu, I simply removed report_result if a server is not pingable (both IPv4 and IPv6) from function cthon_main() as function checkServers() has helped to check the server is pingable. That is, if a server is not available for testing within the logic of cthon_main(), we just skip it.